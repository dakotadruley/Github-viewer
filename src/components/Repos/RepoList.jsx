import React from 'react';
import PropTypes from 'prop-types';
import Repo from './Repo.jsx';

const RepoList = ({ repos }) => {
  const repoElements = repos.map(repo => (
    <li key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <ul>
      {repoElements}
    </ul>
  );
};

RepoList.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default RepoList;
