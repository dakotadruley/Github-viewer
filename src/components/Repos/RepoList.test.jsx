import React from 'react';
import { shallow } from 'enzyme';
import RepoList from './RepoList';

describe('RepoList', () => {
  it('match a snapshot', () => {
    const repos = [
      {
        id: 1,
        name: 'dakota-app',
        url: 'URL1'
      },
      {
        id: 2,
        name: 'dakota-app-2',
        url: 'URL2'
      }
    ];
    const wrapper = shallow(<RepoList repos={repos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
