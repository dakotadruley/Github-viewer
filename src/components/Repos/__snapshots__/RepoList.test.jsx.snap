// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList match a snapshot 1`] = `
<ul>
  <li
    key="1"
  >
    <Repo
      id={1}
      name="dakota-app"
      url="URL1"
    />
  </li>
  <li
    key="2"
  >
    <Repo
      id={2}
      name="dakota-app-2"
      url="URL2"
    />
  </li>
</ul>
`;
